### PR TITLE
8285149: Using HashMap.newHashMap to replace new HashMap(int)

### PR DESCRIPTION
**Pull request by XenoAmess**
These are the changes that too many to be reviewed in 8186958, thus split some of them out.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8285149](https://bugs.openjdk.java.net/browse/JDK-8285149): Using HashMap.newHashMap to replace new HashMap(int)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8301/head:pull/8301` \
`$ git checkout pull/8301`

Update a local copy of the PR: \
`$ git checkout pull/8301` \
`$ git pull https://git.openjdk.java.net/jdk pull/8301/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8301`

View PR using the GUI difftool: \
`$ git pr show -t 8301`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8301.diff">https://git.openjdk.java.net/jdk/pull/8301.diff</a>

</details>

### CODE REVIEW

**Comment by liach** [comment]
Would you consider replacing usages of `new HashMap<>(size, 1)`, such as in `AnnotationInvocationHandler`, as well? Such maps have a smaller table and are more likely to have collisions than `HashMap.newHashMap(size)`.

File: src/demo/share/java2d/J2DBench/src/j2dbench/tests/text/TextTests.java [hunk 1]
```diff
@@ -288,7 +288,7 @@ static String getString(Object key, int len) {
     }
 
 
-    static HashMap strcache = new HashMap(tscripts.length);
+    static HashMap strcache = HashMap.newHashMap(tscripts.length);
```
Comment (mrserb) [inline src/demo/share/java2d/J2DBench/src/j2dbench/tests/text/TextTests.java:291]: This code maintains compatibility with jdk1.4, not necessary to change it.

File: src/java.datatransfer/share/classes/java/awt/datatransfer/SystemFlavorMap.java [hunk 2]
```diff
@@ -812,7 +812,7 @@ public synchronized Map<String,DataFlavor> getFlavorsForNatives(String[] natives
             nativesList.toArray(natives);
         }
 
-        Map<String, DataFlavor> retval = new HashMap<>(natives.length, 1.0f);
+        Map<String, DataFlavor> retval =HashMap.newHashMap(natives.length);
```
Comment (mrserb) [inline src/java.datatransfer/share/classes/java/awt/datatransfer/SystemFlavorMap.java:815]: missed space after '='

File: src/demo/share/java2d/J2DBench/src/j2dbench/tests/text/TextTests.java (hunk 1 above)
Comment (XenoAmess) [reply src/demo/share/java2d/J2DBench/src/j2dbench/tests/text/TextTests.java]: @mrserb done.

File: src/java.datatransfer/share/classes/java/awt/datatransfer/SystemFlavorMap.java (hunk 2 above)
Comment (XenoAmess) [reply src/java.datatransfer/share/classes/java/awt/datatransfer/SystemFlavorMap.java]: @mrserb done.

**Comment by prrace** [comment]
I'm getting a bit tired of seeing these JDK wide changes with lots of files touched.
Delete all changes in desktop from this PR and re-submit them as a separate PR.

Also do not change J2DBench. We deliberately avoid using new API so that we can take it and
run it on very old JDK versions to help track regressions.

**Comment by XenoAmess** [comment]
> I'm getting a bit tired of seeing these JDK wide changes with lots of files touched. 

I'm also a bit tired of seeing so many amazing mis-use codes in jdk. In some place I even see somebody creates a HashMap with factor = 10. I felt my eyes widen and likes seeing a new world or something. Nevermind.

> Delete all changes in desktop from this PR and re-submit them as a separate PR.
> 
> Also do not change J2DBench. We deliberately avoid using new API so that we can take it and run it on very old JDK versions to help track regressions.

Will do 10+ hours later when I off work.

File: src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java [hunk 3]
```diff
@@ -55,7 +55,7 @@ public SunRsaSignEntries(Provider p) {
 
         // start populating content using the specified provider
         // common attribute map
-        HashMap<String, String> attrs = new HashMap<>(3);
+        HashMap<String, String> attrs = HashMap.newHashMap(3);
```
Comment (wangweij) [inline src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java:58]: Looks like 1 is enough.

Comment (XenoAmess) [reply src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java]: > Looks like 1 is enough.

@wangweij done.

**Comment by XenoAmess** [comment]
> @XenoAmess This pull request has been inactive for more than 4 weeks and will be automatically closed if another 4 weeks passes without any activity. To avoid this, simply add a new comment to the pull request. Feel free to ask for assistance if you need help with progressing this pull request towards integration!

comment

**Pull request closed without merging**